frontend/src/metabase/ui/components/overlays/Popover/index.tsx

```diff
@@ -47,10 +52,40 @@ const PopoverDropdown = forwardRef(function PopoverDropdown(
 
 // @ts-expect-error -- our types are better
 PopoverDropdown.displayName = MantinePopoverDropdown.displayName;
+
 // @ts-expect-error -- our types are better
 MantinePopover.Dropdown = PopoverDropdown;
 
-const Popover = MantinePopover;
+type ExtendedPopoverProps = PopoverProps & {
+  // When set to true, allows nested components to customise the
+  // closeOnClickOutside and closeOnEscape behaviour through the
+  // usePreventClosePopover hook.
+  canPreventCloseHandlers?: boolean;
+};
+
+export function Popover({
```

Comment:
@romeovs Is the idea of this to prevent clicking a child popover from closing the parent popover?

Reply:
The idea is that a deeply nested child of the popover needs to have some control over when the popover is allowed to stay open when the user clicks outside it or presses Escape.